#! /bin/sh /usr/share/dpatch/dpatch-run
## avoid-ldap-by-default.dpatch by Ryan Niebur <ryanryan52@gmail.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: by default --avoid-ldap since apache2 is the only user, and we
## don't want to add extra dependencies to other apr-utils rdepends

@DPATCH@
diff -urNad apr-util~/apu-config.in apr-util/apu-config.in
--- apr-util~/apu-config.in	2009-03-27 11:30:29.000000000 -0700
+++ apr-util/apu-config.in	2009-03-27 11:35:15.000000000 -0700
@@ -30,7 +30,8 @@
 LIBS="@APRUTIL_EXPORT_LIBS@"
 INCLUDES="@APRUTIL_INCLUDES@"
 LDFLAGS="@APRUTIL_LDFLAGS@"
-LDAP_LIBS="@LDADD_ldap@"
+ORIG_LDAP_LIBS="@LDADD_ldap@"
+LDAP_LIBS=""
 DBM_LIBS="@LDADD_dbm_db@ @LDADD_dbm_gdbm@ @LDADD_dbm_ndbm@"
 
 APRUTIL_LIBNAME="@APRUTIL_LIBNAME@"
@@ -54,7 +55,7 @@
   --includedir      print location where headers are installed
   --ldflags         print linker flags
   --libs            print library information
-  --avoid-ldap      do not include ldap library information with --libs
+  --avoid-ldap      do not include ldap library information with --libs (default on Debian)
   --ldap-libs       print additional library information to link with ldap
   --srcdir          print APR-util source directory
   --link-ld         print link switch(es) for linking to APR-util
@@ -119,7 +120,7 @@
     flags="$flags $LDAP_LIBS $LIBS"
     ;;
     --ldap-libs)
-    flags="$flags $LDAP_LIBS"
+    flags="$flags $ORIG_LDAP_LIBS"
     ;;
     --includedir)
     if test "$location" = "installed"; then
